### ISSUE TITLE
Passing a python variable to stdin of a shell command?

### ISSUE THREAD

**Issue by Chiel92** [ipython/ipython#9377]
Is it possible to pass a certain python variable to the stdin of some shell command? If not, this is a feature request.

You could of course do something like

```ipython
!printf "$myvar" | some_external_command
```

but there are two major drawbacks here:

1. There is a limit on the argument length of commands, which restricts the size of the input.
2. The input cannot contain a null byte.

**Comment by takluyver** [comment]
No, I don't think we have a way to do that. You may need to find another framework for handling subprocesses, since IPython's `!` shortcut seems to be very limiting for you, and we're not particularly interested in expanding its scope at present.
